web/modules/custom/node_view_count/node_view_count.module

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for node_view_counter module.
+ */
+function node_view_count_entity_view(array &$build, \Drupal\Core\Entity\EntityInterface $entity, \Drupal\Core\Entity\Display\EntityViewDisplayInterface $display, $view_mode) {
+  // Check if the entity is a node.
+  if ($entity->getEntityTypeId() === 'node') {
+    // Get the current session.
+    $session = \Drupal::request()->getSession();
+    // Get the node ID.
+    $nid = $entity->id();
+    // Get the current count from the session.
+    $count = $session->get('node_view_count_' . $nid, 0);
+    // Increment the count.
+    $count++;
+    // Store the updated count back in the session.
+    $session->set('node_view_count_' . $nid, $count);
+    // Add the count to the node render array.
+    $build['view_count'] = [
+      '#markup' => '<p>This node has been viewed ' . $count . ' times in this session.</p>',
+      '#weight' => 100,
+    ];
+    $messenger = \Drupal::messenger();
+    $messenger->addStatus(t('This node has been viewed @count times in this session.', ['@count' => $count]));
+  }
+}
```

Comment:
Could we create a custom hook that will provide this feature to other programmers?